spec/models/loader/types.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe Loader::Types do
+  context "Check user creation verification" do
+    def new_user()
+      Loader::Types::User.new(
+        id: 'alice')
+    end
+    it "should not allow user creation not under root" do
+      ENV['CONJUR_ALLOW_USER_CREATION'] = 'false'
+      user = new_user()
+      status='success'
+      begin
+        user.check_user_creation_allowed(res_id: 'alice@my_sub_tree')
+      rescue Exceptions::InvalidPolicyObject => exc
+        status='failure'
+      ensure
+        ENV['CONJUR_ALLOW_USER_CREATION'] = 'true'
+      end
+      expect(status).to eq('failure')
```

Comment:
We can simplify the above by mocking the environment variable:
```ruby
allow(ENV).to receive(:[]).with('CONJUR_ALLOW_USER_CREATION').and_return('false')
```
and checking for he exception:
```ruby
expect { new_user.check_user_creation_allowed(res_id: 'alice@my_sub_tree') }.to raise_error(Exceptions::InvalidPolicyObject)
```

Reply:
done